flightsql/src/planner.rs

```diff
@@ -364,17 +404,157 @@ async fn plan_get_tables(
 
     let plan = ctx.sql_to_logical_plan(query.as_str()).await?;
     debug!(?plan, "Prepared plan is");
-    Ok(plan.with_param_values(params)?)
+    let plan = plan.with_param_values(params)?;
+
+    if include_schema {
+        add_schema_column_to_plan(ctx, plan).await
+    } else {
+        Ok(plan)
+    }
 }
 
-/// The schema for GetTables
-static GET_TABLES_SCHEMA: Lazy<Schema> = Lazy::new(|| {
-    Schema::new(vec![
+async fn add_schema_column_to_plan(
+    ctx: &IOxSessionContext,
+    plan: LogicalPlan,
+) -> Result<LogicalPlan> {
+    let physical_plan = ctx.create_physical_plan(&plan).await?;
+    let context = ctx.inner().task_ctx();
+
+    // get all the plans output into a Vec<RecordBatch>
+    // like
+    // catalog_name | db_schema_name | table_name | table_type
+    // -------------+----------------+------------+-----------
+    // iox          | public         | foo        | BASE TABLE
+    // iox          | public         | bar        | BASE TABLE
+    let batches = collect(physical_plan, context).await?;
+    if batches.is_empty() {
+        let new_batch = RecordBatch::new_empty(Arc::clone(&GET_TABLES_SCHEMA_WITHOUT_TABLE_SCHEMA));
+        return Ok(ctx.batch_to_logical_plan(new_batch)?);
+    }
+
+    let mut builder = GenericBinaryBuilder::new();
+    // concat multiple batches into a single RecordBatch
+    let batch =
+        match arrow::compute::concat_batches(&GET_TABLES_SCHEMA_WITHOUT_TABLE_SCHEMA, &batches) {
+            Ok(batch) => batch,
+            Err(e) => return Err(Error::Arrow { source: (e) }),
+        };
```

Comment:
This pattern can be written much more concisely like this:

```suggestion
let batch = arrow::compute::concat_batches(&GET_TABLES_SCHEMA_WITHOUT_TABLE_SCHEMA, &batches)?;
```

Which is some rust magic `?` operator: https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html

The `?` basically is shorthand for what you wrote (returns), except that instead of explicating making `Error::Arrow ...` it calls `e.into()` which invokes the "From" implementation:

https://docs.rs/arrow-flight/35.0.0/arrow_flight/error/enum.FlightError.html#impl-From%3CArrowError%3E-for-FlightError

p.s. This was not at all obvious to me when first working in Rust